lib/Tool/Authentication.php

```diff
@@ -57,26 +68,89 @@ public static function authenticateSession(Request $request = null): ?User
             }
         }
 
-        if (!Session::requestHasSessionId($request, true)) {
-            // if no session cookie / ID no authentication possible, we don't need to start a session
+        $session = $request->hasPreviousSession() ? $request->getSession() : null;
+
+        if (null === $session) {
             return null;
         }
 
-        $session = Session::getReadOnly();
-        $user = $session->get('user');
+        $token = $session->get('_security_pimcore_admin');
+        $token = static::safelyUnserialize($token);
```

Comment:
```suggestion
        $token = $token ? static::safelyUnserialize($token) : null;
```
`token` can be null if `authenticateSession` called outside admin session scope. please [see ](https://github.com/pimcore/pimcore/blob/83142b53cfd5a791e3c6167470a66336fd03da22/lib/Targeting/EventListener/ToolbarListener.php#L139)

Reply:
done